### PR TITLE
Use the injected treeProps for Layout PerfEvent

### PR DESCRIPTION
**Pull request by boduan402**
At the runtime, the ComponentContext's treeProps could be null. In this case, it would better to use the treeProps that provided by caller.

### CODE REVIEW

File: litho-core/src/main/java/com/facebook/litho/LogTreePopulator.java [hunk 1]
```diff
@@ -38,6 +38,21 @@ private LogTreePopulator() {}
   @CheckReturnValue
   public static PerfEvent populatePerfEventFromLogger(
       ComponentContext c, ComponentsLogger logger, PerfEvent perfEvent) {
+    return populatePerfEventFromLogger(c, c.getTreeProps(), logger, perfEvent);
+  }
+
+  /**
+   * Annotate a log event with the log tag set in the context, and extract the treeprops from a
```
Comment (colriot) [inline litho-core/src/main/java/com/facebook/litho/LogTreePopulator.java]: Seems that jdoc was just copy-pasted without modification

File: litho-core/src/main/java/com/facebook/litho/ComponentTree.java [hunk 2]
```diff
@@ -1772,7 +1772,10 @@ private void calculateLayout(
     final PerfEvent layoutEvent =
         logger != null
             ? LogTreePopulator.populatePerfEventFromLogger(
-                mContext, logger, logger.newPerformanceEvent(mContext, EVENT_LAYOUT_CALCULATE))
+                mContext,
+                mContext.getTreeProps() == null ? treeProps : mContext.getTreeProps(),
```
Comment (colriot) [inline litho-core/src/main/java/com/facebook/litho/ComponentTree.java]: According to the nearby usage, `context.getTreeProps` is used as a fallback. Let's utilize the same logic here: just passing `treeProps` in, and falling back in overridden `populatePerfEventFromLogger` method.

File: litho-core/src/main/java/com/facebook/litho/LogTreePopulator.java (hunk 1 above)
Comment (boduan402) [reply litho-core/src/main/java/com/facebook/litho/LogTreePopulator.java]: Thanks Sergey, updated.

File: litho-core/src/main/java/com/facebook/litho/ComponentTree.java (hunk 2 above)
Comment (boduan402) [reply litho-core/src/main/java/com/facebook/litho/ComponentTree.java]: Good point. updated.

**Comment by boduan402** [comment]
I mess up the forked branch. Will create a new one.

**Pull request closed without merging**